testing/python/collect.py

```diff
@@ -705,6 +711,7 @@ def test_xfail(x):
         result = testdir.runpytest()
         result.stdout.fnmatch_lines('* 2 passed, 1 xpassed in *')
 
+    @pytest.mark.filterwarnings('ignore:Applying marks directly to parameters')
```

Comment:
> those mark applications are all the same, can we assign a speaking name to that?

Good idea, will do.

> I wonder if those tests should just be using pytest.param instead?

I think we should keep them in to ensure the old method still works, so we should remove them only when we remove the functionality.